src/main/java/com/hubspot/jinjava/lib/filter/AbstractFilter.java

```diff
@@ -0,0 +1,189 @@
+      if (argName == null) {
+        throw new InvalidInputException(
+                interpreter,
+                "INVALID_ARG_NAME",
+                String.format("Argument at index '%s' is invalid for filter %s", i, getName())
+        );
+      }
+      namedArgs.put(argName, arg);
+    }
+
+    //Parse args based on their declared types
+    Map<String, Object> parsedArgs = new HashMap<>();
+    namedArgs.forEach((k, v) -> parsedArgs.put(k, parseArg(interpreter, validNamedArgs.get(k), v)));
+
+    validateArgs(interpreter, parsedArgs);
+
+    return filter(var, interpreter, parsedArgs);
+  }
+
+  public Object parseArg(JinjavaInterpreter interpreter, JinjavaParam jinjavaParamMetadata, Object value) {
+    if (jinjavaParamMetadata.type() == null ||
+        Arrays.asList("object", "dict", "sequence").contains(jinjavaParamMetadata.type())) {
+      return value;
+    }
+    String valueString = Objects.toString(value, null);
+    switch (jinjavaParamMetadata.type().toLowerCase()) {
+      case "boolean": {
+        return BooleanUtils.toBoolean(valueString);
+      }
+      case "int": {
+        return NumberUtils.toInt(valueString);
+      }
+      case "long": {
+        return NumberUtils.toLong(valueString);
+      }
+      case "float": {
+        return NumberUtils.toFloat(valueString);
+      }
+      case "double": {
+        return NumberUtils.toDouble(valueString);
+      }
+      case "number": {
+        return new BigDecimal(valueString);
+      }
+      case "string": {
+        return valueString;
+      }
+    }
+    throw new InvalidInputException(
+            interpreter,
+            "INVALID_ARG_NAME",
+            String.format("Argument named '%s' with value '%s' cannot be parsed for filter %s", jinjavaParamMetadata.value(), getName())
+    );
+  }
+
+  public void validateArgs(JinjavaInterpreter interpreter, Map<String, Object> parsedArgs) {
+    for(JinjavaParam jinjavaParam: getNamedArguments().values()) {
+      if(jinjavaParam.required() && !parsedArgs.containsKey(jinjavaParam.value())) {
+        throw new InvalidInputException(
+                interpreter,
+                "MISSING_REQUIRED_ARG",
+                String.format("Argument named '%s' is required but missing for filter %s", jinjavaParam.value(), getName())
+        );
+      }
+    }
+  }
+
+  public Map<String, JinjavaParam> getNamedArguments() {
+    if (isNamedArgumentCheckDone) {
+      return namedArguments;
+    }
+    JinjavaDoc jinjavaDoc = this.getClass().getAnnotation(JinjavaDoc.class);
+    if (jinjavaDoc != null) {
+      namedArguments = new LinkedHashMap<>();
+      for (JinjavaParam jinjavaParam: jinjavaDoc.params()) {
+        namedArguments.put(jinjavaParam.value(), jinjavaParam);
+      }
+      namedArguments = Collections.unmodifiableMap(namedArguments);
+    }
+    isNamedArgumentCheckDone = true;
+    return namedArguments;
+  }
+
+  public int getNamedArgumentPosition(String argName) {
+    getNamedArguments();
```

Comment:
You could rewrite this in funcional style with something like

`return Optional.ofNullable(getNamedArguments()).map(a -> a.keySet()).filter(argNames -> a.contains(argName)).map(a -> a.indexOf(argName)).orElse(-1);`